### PR TITLE
Add local score function

### PR DESCRIPTION
**Pull request by wangxiyuan**
1. fix package issue
2. Do not use `assert`
3. Add two metric: `code line change in recent one  year`, `active contributor in recent one year (commit  >=20)`
4. Add a new option `repos_location ` to specify locaion git repo folder. The repo should be in the place like `/opt/repos/tensorflow/tensorflow`
5. change `--projects_list` and `--result_file` to `--project-list` and `--result-file` for human usage.
6. Update CI to support local score

### CODE REVIEW

File: etc/reposcore.conf [hunk 1]
```diff
@@ -1,5 +1,6 @@
 [global]
 retry = 3
+repos_location = /opt/repos
```
Comment (ZhengZhenyu) [inline etc/reposcore.conf:5]: Add a comment for this parameter, for example, "Location of the local repos"

File: reposcore/cli.py [hunk 2]
```diff
@@ -56,6 +56,10 @@ def _init_constants(self):
         cs_run.DEPENDENTS_COUNT_WEIGHT = float(
             self.config.get('weight', 'dependents_count_weight'))
 
+        cs_run.PARAMS.append("code_line_change_recent_year")
+        cs_run.PARAMS.append("activity_member_count_recent_year")
```
Comment (ZhengZhenyu) [inline reposcore/cli.py]: member seems misleading to me

File: reposcore/repo/repo.py [hunk 3]
```diff
@@ -19,15 +20,71 @@
...
+            if 'files changed' not in change:
+                continue
+            if 'insertions' in change:
+                addition += int(change.split(' insertions')[0].split(', ')[-1])
+            if 'deletions' in change:
+                deletion += int(change.split(' deletions')[0].split(', ')[-1])
+        return "+%d, -%d" % (addition, deletion)
+
+    @property
+    def activity_member_count_recent_year(self):
```
Comment (ZhengZhenyu) [inline reposcore/repo/repo.py]: ditto

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -32,4 +32,6 @@ jobs:
       - name: Run
         run: |
           export GITHUB_AUTH_TOKEN=${{ secrets.GITHUB_TOKEN }}
-          reposcore --projects_list test/ci-test.txt --result_file test/result.csv
+          mkdir -p /opt/repos/kunpengcompute
+          git clone https://github.com/kunpengcompute/reposcore /opt/repos/kunpengcompute/reposcore
```
Comment (Yikun) [inline .github/workflows/main.yml:36]: clone的过程应该放在reposcore里面还是外面？

File: reposcore/repo/repo.py [hunk 5]
```diff
@@ -19,15 +20,74 @@
...
+        except Exception:
+            raise Exception("No local git repo find: %s" % repo_location)
+        self.since_time = self._get_start_date()
+
+    def _get_start_date(self):
+        start_year = int(time.strftime('%Y',time.localtime(time.time()))) - 1
+        month_day = time.strftime('%m-%d',time.localtime(time.time()))
+        return '{}-{}'.format(start_year,month_day)
+
+    def _name_fix(self, author_raw):
```
Comment (Yikun) [inline reposcore/repo/repo.py:36]: 这种typo咋发现的？发现一个就要转换一个？

File: test/ci-test.txt [hunk 6]
```diff
@@ -1,5 +1 @@
-https://github.com/tensorflow/tensorflow
```
Comment (Yikun) [inline test/ci-test.txt:1]: 误操作还是觉得reposcore就够了？最好保留>=2行比较好

File: reposcore/cli.py [hunk 7]
```diff
@@ -56,17 +56,21 @@ def _init_constants(self):
...
         parser.add_argument('-c', dest='config',
                             help='path to config file')
-        parser.add_argument("--projects_list",
+        parser.add_argument("--project-list",
                                  type=open,
                                  required=True,
                                  help="File name of projects url list.")
-        parser.add_argument("--result_file",
+        parser.add_argument("--result-file",
                                  type=str,
```
Comment (Yikun) [inline reposcore/cli.py:74]: btw，这个其实给个默认值（比如result.csv）比较好。这样用户上手的时候，只需要给个列表的文件了

File: .github/workflows/main.yml (hunk 4 above)
Comment (wangxiyuan) [reply .github/workflows/main.yml:36]: 目前是手动clone，repo还没加自动clone功能。后续可以加，感觉必要性不是特别大，再讨论吧

File: reposcore/cli.py (hunk 7 above)
Comment (wangxiyuan) [reply reposcore/cli.py:74]: +1， TODO

File: reposcore/repo/repo.py (hunk 5 above)
Comment (wangxiyuan) [reply reposcore/repo/repo.py:36]: 是的，目前只能这样。这里应该加个try cache，后续遇到问题了能直接知道问题名字是啥 TODO

File: test/ci-test.txt (hunk 6 above)
Comment (wangxiyuan) [reply test/ci-test.txt:1]: 为了CI更快一点，只clone了reposcore，再加一个也行，你看加哪个，需要和CI同步修改

**Comment by Yikun** [comment]
LGTM, nits could be addressed in followup

**Pull request merged**